### DISCUSSION TITLE
Does Quarkus have the spring4shell vulnerability?

### DISCUSSION THREAD

**Discussion by HeeroJheng** [quarkusio/quarkus#25398] [Q&A] [answered]
Hi,

We developed microservices with Quarkus and its extensions for Spring( DI, Web API, and etc). Does it have any spring4shell vulnerability in our microservices?

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
Hi, Quarkus does not suffer from said vulnerability because it never executes any Spring classes, it just uses the Spring API.

If you dig even deeper, you'll see that the problematic Spring classes are not even on the classpath

**Reply by HeeroJheng** [reply]
Thanks for your answer!

**Reply by geoand** [reply]
YW!
